project-koku/koku
# COST-3050 Repeated update of timestamp for OCI source

## Jira Ticket

[COST-3050](https://issues.redhat.com/browse/COST-3050)

## Description

This change updates OCI downloaders to create separate manifests for each month's data ingested

## Testing

1. Checkout Branch
2. Restart Koku and 
3. OCI Local Downloader; load data with `make load-test-customer-data test_source=oci`
4. Hit endpoint once everything loads, 
     i. http://127.0.0.1:8000/api/cost-management/v1/sources/<source_uuid>/stats/
    ii. You should see more than one month's manifest similar to this

        {
            "provider_linked": true,
            "data_updated_date": "2022-10-27 13:48:22",
            "ocp_on_cloud_data_updated_date": null,
            "2022-10-01": {
                "manifests": [
                    {
                        "assembly_id": "5af2509d-ec07-4907-b305-3b24fd42bcda:2022-10",
                        "billing_period_start": "2022-10-01",
                        "files_processed": "2/2",
                        "last_process_start_date": "2022-10-27 13:48:01",
                        "last_process_complete_date": "2022-10-27 13:48:03",
                        "last_manifest_complete_date": "2022-10-27 13:48:22",
                        "manifest_modified_datetime": "2022-10-27 13:47:41"
                    }
                ]
            },
            "2022-09-01": {
                "manifests": [
                    {
                        "assembly_id": "5af2509d-ec07-4907-b305-3b24fd42bcda:2022-09",
                        "billing_period_start": "2022-09-01",
                        "files_processed": "2/2",
                        "last_process_start_date": "2022-10-27 13:47:53",
                        "last_process_complete_date": "2022-10-27 13:47:55",
                        "last_manifest_complete_date": "2022-10-27 13:48:22",
                        "manifest_modified_datetime": "2022-10-27 13:47:41"
                    }
                ]
            },
            "2022-08-01": {
                "manifests": [
                    {
                        "assembly_id": "5af2509d-ec07-4907-b305-3b24fd42bcda:2022-08",
                        "billing_period_start": "2022-08-01",
                        "files_processed": "2/2",
                        "last_process_start_date": "2022-10-27 13:47:43",
                        "last_process_complete_date": "2022-10-27 13:47:46",
                        "last_manifest_complete_date": "2022-10-27 13:48:22",
                        "manifest_modified_datetime": "2022-10-27 13:47:41"
                    }
                ]
            }
        }


    
 
5. Testing the OCI real downloader requires OCI credentials and a real source  👈🏿  happy to help with this
     - make OCI source
     ```
      make oci-source
      ```
      - Hit endpoint in `4. i` and you should see a similar response as in `4. ii`

## Notes

...
